apps/open-swe/langbench/utils.ts

```diff
@@ -0,0 +1,227 @@
+import { createLogger, LogLevel } from "../src/utils/logger.js";
+import { ENV_CONSTANTS } from "../src/utils/env-setup.js";
+import { TestResults, PytestJsonReport, RunPytestOptions } from "./types.js";
+import { readFile } from "../src/utils/read-write.js";
+
+const logger = createLogger(LogLevel.DEBUG, "Langbench Utils");
+
+/**
+ * Fetch diff content from a diff URL and extract test file names, this function is used in one-off situtations to get the test files from the diff url.
+ *
+ */
+export async function getTestFilesFromDiff(diffUrl: string): Promise<string[]> {
+  try {
+    const response = await fetch(diffUrl);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch diff: ${response.statusText}`);
+    }
+
+    const diffContent = await response.text();
+    const testFiles: string[] = [];
+
+    // Parse the diff to find modified files
+    const lines = diffContent.split("\n");
+    for (const line of lines) {
+      // Look for diff file headers
+      if (line.startsWith("diff --git ")) {
+        const match = line.match(/diff --git a\/(.+?) b\//);
+        if (match) {
+          const filePath = match[1];
+          // Check if this is a test file in libs/langgraph/tests/
+          if (isLangGraphTestFile(filePath)) {
+            testFiles.push(filePath);
+          }
+        }
+      }
+    }
+
+    return [...new Set(testFiles)]; // Remove duplicates
+  } catch (error) {
+    logger.error(`Failed to fetch or parse diff from ${diffUrl}:`, { error });
+    return [];
+  }
+}
+
+/**
+ * Check if a file path represents a test file in libs/langgraph/tests/
+ */
+export function isLangGraphTestFile(filePath: string): boolean {
+  return filePath.includes("libs/langgraph/tests/") && filePath.endsWith(".py");
+}
+
+// Use shared constants from env-setup utility
+const { RUN_PYTHON_IN_VENV, RUN_PIP_IN_VENV } = ENV_CONSTANTS;
+
+// Installation commands for pytest and dependencies
+const PYTEST_INSTALL_COMMANDS = [
+  `${RUN_PIP_IN_VENV} install pytest pytest-mock pytest-asyncio syrupy pytest-json-report`,
+  `${RUN_PIP_IN_VENV} install -e ./libs/langgraph`,
+];
+
+/**
+ * Run pytest on specific test files and return structured results
+ */
+export async function runPytestOnFiles(
+  options: RunPytestOptions,
+): Promise<TestResults> {
+  const { sandbox, testFiles, repoDir, timeoutSec = 300 } = options;
+  if (testFiles.length === 0) {
+    logger.warn("No test files provided, skipping pytest execution");
+    return {
+      success: true,
+      error: null,
+      totalTests: 0,
+      passedTests: 0,
+      failedTests: 0,
+      testDetails: [],
+    };
+  }
+
+  logger.info(`Running pytest on ${testFiles.length} test files`, {
+    testFiles,
+  });
+
+  // Join test files for pytest command
+  const testFilesArg = testFiles.join(" ");
+  const command = `${RUN_PYTHON_IN_VENV} -m pytest ${testFilesArg} -v --tb=short --json-report --json-report-file=/tmp/pytest_report.json`;
```

Comment:
```suggestion
  const runTestCommand = `${RUN_PYTHON_IN_VENV} -m pytest ${testFilesArg} -v --tb=short --json-report --json-report-file=/tmp/pytest_report.json`;
```

You have a var also defined as `command` below (`L93`). want to avoid duplicates (separate: we should enable the eslint rule that errors on stuff like this. idk the name but im sure there is one)